Make message-based checks invariant to Locale

msteiger opened the pull request:
This PR uses the `ROOT` locale for tests, rather than `ENGLISH` to fix compatibility issues on non-English systems.

The problem with `ENGLISH` is that Java searches for the file `messages_en.properties`, which does not exist. Instead of falling back to `messages.properties`, it tries to load the one for the current locale first. I'm not sure why it does that.

In any case, the correct way is to use `ROOT`, which explicitly uses the "default" and language-invariant file without language specifier.

**How to reproduce:** Set the default locale to some other language: 

    Locale.setDefault(Locale.GERMAN);

in a unit test that compares text messages and it will fail.

**Note**: A few checks (about 25) are based on comparing Exception messages. They are not affected by this and thus still do not work on non-English Locales.

romani commented:
1) please update commit message with link to this PR -  "...... . Issue #1044" . Code change is not obvious and we might need to come back to your issue description later.

2) Additionally It would be good if you describe, how you run into that problem in PR description.

3) can we make a UT on that ?

msteiger commented:
I updated the commit message and added steps to reproduce in the PR description.

I believe that this is covered already enough. Hundreds of Unit Tests fail if the Locale is incorrect.

romani commented:
Unittest are not failing without your change and with your change, good sign that smth missed in testing. Locale probles are the most complicated as enginees rarely change them.

I would rather have a test that will faill if smb in future decide to return to English hardcoded locale.

msteiger commented:
I guess you could set the default locale to some other language than English, run a few tests and compare the reported messages to the expected ones in English. This should do the trick.

I assume that the core dev. team would know how to implement such a test.

romani commented:
merged as FF
